### DISCUSSION TITLE
How do I get graphQL dgs to stop passing custom scalars as Strings?

### DISCUSSION THREAD

**Discussion by ZMobile** [Netflix/dgs-framework#1323] [Q&A]
When I query fields from this object with MonetaryAmount as a custom scalar:

type PricingDetails {
    id: ID!
    creationTimestamp: LocalDateTime!
    modifiedTimestamp: LocalDateTime!
    isPricedByOrderType: Boolean!
    unifiedListPrice: MonetaryAmount
    deliveryListPrice: MonetaryAmount
    pickupListPrice: MonetaryAmount
    overTheCounterListPrice: MonetaryAmount
}

scalar MonetaryAmount

This is what I get as an http response:

"pricingDetails":{
                  "isPricedByOrderType":false,
                  "unifiedListPrice":"{\"amount\":0,\"currency\":\"USD\"}",
                  "deliveryListPrice":"{\"amount\":0,\"currency\":\"USD\"}",
                  "pickupListPrice":"{\"amount\":0,\"currency\":\"USD\"}",
                  "overTheCounterListPrice":"{\"amount\":0,\"currency\":\"USD\"}"
               }

I'm not sure why its passing my custom scalars as strings. Is there a way to disable this? It passes the correct object, just surrounded by an extra set of quotation marks. I feel like there must be some "pass custom scalars as string" configuration you can turn off.

**Comment by srinivasankavitha** [comment]
Have you tried adding a scalar implementation for MonetaryAmount?
https://netflix.github.io/dgs/scalars/
This should specify how you want the scalar to be serialized/deserialized.

**Reply by ZMobile** [reply]
Yes  I should've probably included it. Here is a copy of that:

`import com.google.gson.Gson;
import com.netflix.graphql.dgs.DgsScalar;
import graphql.schema.Coercing;
import graphql.schema.CoercingSerializeException;

import javax.money.MonetaryAmount;

@DgsScalar(name = "MonetaryAmount")
public class MonetaryAmountScalar implements Coercing<MonetaryAmount, String> {
    private final Gson gson;

    public MonetaryAmountScalar(Gson gson) {
        this.gson = gson;
    }

    @Override
    public String serialize(Object input) {
        if (input instanceof MonetaryAmount) {
            MonetaryAmount monetaryAmount = (MonetaryAmount) input;
            return gson.toJson(monetaryAmount);
        }

        throw new CoercingSerializeException("Not a valid MonetaryAmount");
    }

    @Override
    public MonetaryAmount parseValue(Object input) {
        return gson.fromJson((String) input, MonetaryAmount.class);
    }

    @Override
    public MonetaryAmount parseLiteral(Object input) {
        return gson.fromJson((String) input, MonetaryAmount.class);
    }
}`

Strangely enough this didn't really explain why it was passing these objects with an extra set of quotation marks. Once I get around to it I might try trimming the beginning and end of the output of gson.toJson() there (maybe theres one more set of quotation marks there than there should be) but that seems like an odd solution anyways.

I did manage to get a temporary fix by having the json parser on the client side factor for potential "String-objects" but I still didn't manage to figure out why dgs passes this as a string rather than an object.
